traffic_ops/client/session.go

```diff
@@ -207,6 +244,36 @@ func LoginWithAgent(toURL string, toUser string, toPasswd string, insecure bool,
 	return to, remoteAddr, nil
 }
 
+func LoginWithToken(toURL string, token string, insecure bool, userAgent string, useCache bool, requestTimeout time.Duration) (*Session, net.Addr, error) {
+	options := cookiejar.Options {
+		PublicSuffixList: publicsuffix.List,
+	}
+
+	jar, err := cookiejar.New(&options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	client := http.Client {
+		Timeout: requestTimeout,
+		Transport: &http.Transport {
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: insecure},
```

Comment:
do we use insecure on all our api tests?

Reply:
no, that's based on the value of the parameter given by the caller. So each program that uses the client provides the choice of whether or not to skip cert verification.